graphqlmetrics/core/server.go

```diff
@@ -32,21 +44,53 @@ func NewServer(metricsService graphqlmetricsv1connect.GraphQLMetricsServiceHandl
 		opt(s)
 	}
 
-	s.bootstrap()
+	s.bootstrap(ctx)
 
 	return s
 }
 
-func (s *Server) bootstrap() {
+func (s *Server) bootstrap(ctx context.Context) {
 	mux := http.NewServeMux()
+	var interceptors = []connect.Interceptor{}
+
+	if s.metricConfig.IsEnabled() {
+		mp, registry, err := s.metricConfig.NewPrometheusMeterProvider(ctx)
+		if err != nil {
+			s.logger.Error("Failed to create Prometheus exporter", zap.Error(err))
+		}
+		s.meterProvider = mp
+		s.registry = registry
+
+		baseAttributes := []attribute.KeyValue{}
+
+		metricsStore, err := telemetry.NewPromMetricStore(s.logger, s.meterProvider, baseAttributes)
+		if err != nil {
+			s.logger.Error("Failed to create metric store", zap.Error(err))
+		}
+
+		s.metricConfig.MetricStore = metricsStore
+	}
+
+	if s.metricConfig.Prometheus.Enabled {
+		s.prometheusServer = telemetry.NewPrometheusServer(s.logger, s.metricConfig.Prometheus.ListenAddr, s.metricConfig.Prometheus.Path, s.registry)
+		interceptors = append(
+			interceptors,
+			s.metricConfig.ObservabilityInterceptor(),
+		)
+	}
+
 	path, handler := graphqlmetricsv1connect.NewGraphQLMetricsServiceHandler(
 		s.metricsService,
 		// Brotli compression support.
 		brotli.WithCompression(),
+		connect.WithInterceptors(interceptors...),
 	)
-	mux.Handle("/health", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+
+	healthHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		w.WriteHeader(http.StatusOK)
-	}))
+	})
+
+	mux.Handle("/health", healthHandler)
 	mux.Handle(path, authenticate(s.jwtSecret, s.logger, handler))
```

Comment:
Unfortunately, I have seen this too late. Instead of using an interceptor we can track the metrics in a regular handler. We would also get access to the JWT which contains the organization and federated graph id. That would make the metrics much more insightful.

Reply:
Wouldn't it be possible to pass the parent context to the interceptor and the pull out the same information?

Reply:
Yes, when you get access to the request context.

Reply:
Works with the interceptor. 

The claims could be unset, this shouldn't happen as the  validation happens already before, but in case the claims are empty, I pass back a `INVALID_ARGUMENT`. Let me know about your opinion.

Reply:
Sounds good